runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -223,7 +223,11 @@ public void start() throws Exception
             AgentRunner.startOnThread(runner, Thread::new);
         }
         watcherTaskRef = commonPool().submit(watcherTask);
-        watcherTask.doInitialConfiguration(rootConfigURL);
+        NamespaceConfig initialConfig = watcherTask.watch(rootConfigURL).get();
+        if (initialConfig == null)
+        {
+            throw new Exception("Parsing of the initial configuration failed.");
+        }
```

Comment:
This is good, however I was thinking the exception would be set on the future internally by the watcher task, for example via `CompletableFuture.failedFuture(ex)`, rather than returning null and converting it at the caller, then you just need to call `.get()` and the exception will be thrown for you if unsuccessful.